File: src/main/java/org/example/queries/MovingItemDTOImpl.java
package org.example.queries;

import org.example.api.MovingItemDTO;

public class MovingItemDTOImpl implements MovingItemDTO {
    String name;
    int[] location;
    int moves;
    int value;


    public MovingItemDTOImpl(String name, int[] location, int moves, int value) {
        this.name = name;
        this.location = location;
        this.moves = moves;
        this.value = value;
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public int[] getLocation() {
        return location;
    }

    @Override
    public int getNumberOfMoves() {
        return moves;
    }

    @Override
    public int getValue() {
        return value;
    }
}

File: src/main/kotlin/query/impl/MovingItemDTOImpl.kt
package query.impl

import command.api.MovingItem
import query.api.MovingItemDTO

class MovingItemDTOImpl(
    override val name: String,
    override val location: IntArray,
    override val moves: Int,
    override val value: Int
) : MovingItemDTO {

    companion object {
        fun fromMovingItem(item: MovingItem) = MovingItemDTOImpl(item.name, item.location, item.moves, item.value)
    }
}